app/controllers/claims_controller.rb

```diff
@@ -297,4 +304,19 @@ def save_details_check
     details_check = params.dig(:claim, :details_check)
     DfeIdentity::ClaimUserDetailsCheck.call(current_claim, details_check)
   end
+
+  def update_session_with_tps_school
+    if (school = current_claim.recent_tps_school)
+      session[:tps_school_id] = school.id
+      session[:tps_school_name] = school.name
+      session[:tps_school_address] = school.address
+    end
+  end
+
+  def check_correct_school_params
+    updated_claim_params = claim_params
+    CorrectSchoolFormChecker.call(updated_claim_params, change_school: params[:change_school])
```

Comment:
Does this mutate the `updated_claim_params`?

Reply:
I was dubious as well...

```
irb(main):005:0> p = {}
=> {}
irb(main):006:0> CorrectSchoolFormChecker.call(p, change_school: "true")
=> {:current_school_id=>nil, :school_somewhere_else=>true}
irb(main):007:0> p
=> {:eligibility_attributes=>{:current_school_id=>nil, :school_somewhere_else=>true}}
```

Also...

`spec/forms/correct_school_form_checker_spec.rb`

Proves it does mutate it.

Thoughts?

Reply:
I saw the test too, but maybe the fact that we both questioned "does that work like that" it could be refactored a little for readability, maybe. Either way, it is not blocking.